### ISSUE TITLE
Accelerate training not working with PEFT from `adapter-hub/adapter-transformers` but work with `huggingface/peft`

### ISSUE THREAD

**Issue by hepengfe** [huggingface/accelerate#1343]
### System Info

```Shell
- `Accelerate` version: 0.18.0
- Platform: Linux-5.15.0-60-generic-x86_64-with-glibc2.17
- Python version: 3.8.16
- Numpy version: 1.24.2
- PyTorch version (GPU?): 1.13.0+cu116 (True)
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: DEEPSPEED
        - mixed_precision: no
        - use_cpu: False
        - num_processes: 3
        - machine_rank: 0
        - num_machines: 1
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - deepspeed_config: {'gradient_accumulation_steps': 6, 'offload_optimizer_device': 'none', 'offload_param_device': 'none', 'zero3_init_flag': False, 'zero3_save_16bit_model': False, 'zero_stage': 3}
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

I have same training script using acclerate for both packages but accelerate doesn't work with `adapter-hub/adapter-transformers`. I tried training `lora` module.  I have the following observation:
1. training loss is going down to a certain point such 1 and no less loss.
2. it generates empty string after training.
3. parameter does change during training.

Then I checked parameter state dictionary between `adapter-transformers` and `peft` . I think it could be the reason the module is not trained. 
![image](https://user-images.githubusercontent.com/28021889/233539080-68304946-767a-4ec5-bfc3-8caa72fdc642.png)
The image above includes lora modules from packages. 
`lora_A` and `lora_B` come from `peft` package (training working), while `(loras): ModuleDict()` comes from `adapter-transformers` (training not working). I think their different ways of wrapping the two matrices in the model state might causes loss is not backpropagate to `(loras): ModuleDict()` somehow after `accelerate` optimize the optimizer/model.

Based on my observation, I would like some suggestions or ideas about the issues from insiders' view.

### Expected behavior

It's expected to train effectively for PEFT modules implemented by `adapter-hub/adapter-transformers` and generate non-empty string.

**Comment by hepengfe** [comment]
related issue https://github.com/adapter-hub/adapter-transformers/issues/511

**Comment by hepengfe** [comment]
problem solved and it's the issue in my own script (not initialize lm head correctly).
